Repositories/TimeTable.php

```diff
@@ -98,67 +99,72 @@ public function getTimesheetByTicketIdAndWorkDate(string $ticketId, CarbonImmuta
     }
 
     /**
-     * updateTime - update specific time entry
+     * updateOrAddTimelogOnTicket - Updates or adds a timelog entry for a ticket
      *
-     * @param array<string, mixed> $values
+     * @param array $values An array containing the values for the timelog entry
+     * @param int|null $originalId (Optional) The original timelog id to check for updates or deletion
      *
      * @return void
+     * @access public
      */
-    public function updateTime(array $values): void
+    public function updateOrAddTimelogOnTicket(array $values, int $originalId = null): void
     {
-        $sql = 'UPDATE
-                    zp_timesheets
-                SET
-                    hours = :hours,
-                    description = :description,
-                    workDate = :date
-                WHERE
-                    id = :id';
+        $sql = 'SELECT * FROM zp_timesheets WHERE ticketId = :ticketId AND workDate = :date';
+
         $stmn = $this->db->database->prepare($sql);
-        $stmn->bindValue(':hours', $values['hours']);
-        $stmn->bindValue(':description', $values['description']);
+        $stmn->bindValue(':ticketId', $values['ticketId']);
         $stmn->bindValue(':date', $values['workDate']);
-        $stmn->bindValue(':id', $values['id']);
-
         $stmn->execute();
-        $stmn->closeCursor();
-    }
 
-    /**
-     * updateTime - update specific time entry
-     *
-     * @param array<string, mixed> $values
-     *
-     * @return void
-     */
-    public function logTimeOnTicket(array $values): void
-    {
-        $sql = 'INSERT INTO zp_timesheets (
-            userId,
-            ticketId,
-            workDate,
-            hours,
-            description,
-            kind
-        ) VALUES (
-            :userId,
-            :ticket,
-            :date,
-            :hours,
-            :description,
-            :kind
-        )';
-
-        $stmn = $this->db->database->prepare($sql);
-
-        $stmn->bindValue(':userId', $values['userId'], PDO::PARAM_INT);
-        $stmn->bindValue(':ticket', $values['ticketId']);
-        $stmn->bindValue(':date', $values['workDate']);
-        $stmn->bindValue(':kind', $values['kind']);
-        $stmn->bindValue(':description', $values['description']);
-        $stmn->bindValue(':hours', $values['hours']);
+        $timesheet = $stmn->fetch(PDO::FETCH_ASSOC);
+        $stmn->closeCursor();
 
+        if ($timesheet) {
+            if ($originalId && $originalId == $timesheet['id']) {
+                $sql = 'UPDATE zp_timesheets SET hours = :hours, description = :description WHERE id = :id';
```

Comment:
Can this remove an already existing description?

Reply:
Yes, but only if the user clears the description field before saving